app/facades/websocket/WebSocketController.scala

```diff
@@ -174,14 +199,37 @@ class WebSocketController @Inject() (actorSystem:  ActorSystem,
       Future.successful(Left[Result, DSAFlow](Forbidden))
   }
 
-  private def acceptOrResult(f: Option[DSLinkSessionInfo] => Future[Either[Result, Flow[DSAMessage, DSAMessage, _]]]): WebSocket = {
+  private def acceptOrResult(f: Option[DSLinkSessionInfo] => Future[Either[Result, DSAFlow]]): WebSocket = {
     WebSocket { request =>
       log.debug(s"WS request received: $request")
       val dsId = getDsId(request)
+      val clientAuth = getAuth(request)
       val sessionInfo = cache.get[DSLinkSessionInfo](dsId)
       log.debug(s"Session info retrieved for $dsId: $sessionInfo")
 
-      f(sessionInfo).map(_.right.map(getTransformer(sessionInfo).transform))
+      val res =
+        if (validateAuth(sessionInfo, clientAuth))
+          f(sessionInfo)
+        else {
+          log.warn(s"Authentication failed, auth value is not correct.")
```

Comment:
Perhaps, should be log.error(...)? Also, worth to include the details - the dsId, etc. - for forensics.

Reply:
Changed, please check